legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-connection/src/main/java/org/finos/legend/connection/jdbc/JDBCConnectionManager.java

```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.connection.jdbc;
+
+import org.eclipse.collections.impl.map.mutable.ConcurrentHashMap;
+
+import java.util.List;
+import java.util.Properties;
+import java.util.ServiceLoader;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * TODO?: @akphi - This is a temporary hack!
+ * We probably need to have a mechanism to control the connection pool
+ * We cloned DatabaseManager from relational executor, we should consider if we can eventually unify these 2
+ */
+public abstract class JDBCConnectionManager
```

Comment:
Let's refactor connection manager to implement connection pooling and then we can see how it fits (or does not fit) with core relational

Reply:
it would duplicate a fair chunk of code, you're alright with it?